src/audio/asrc/asrc.c

```diff
@@ -859,8 +859,35 @@ static int asrc_reset(struct processing_module *mod)
 	return 0;
 }
 
+#if CONFIG_IPC_MAJOR_3
+static int asrc_init_shim(struct processing_module *mod)
+{
+	struct ipc_config_asrc *ipc_asrc;
+	struct module_data *mod_data = &mod->priv;
+	const struct sof_ipc_comp_asrc *asrc =
+		(const struct sof_ipc_comp_asrc *)mod_data->cfg.init_data;
+
+	if (IPC_TAIL_IS_SIZE_INVALID(*asrc))
+		return -EBADMSG;
+
+	ipc_asrc = rballoc(0, SOF_MEM_CAPS_RAM, sizeof(*ipc_asrc));
```

Comment:
is the configuration so large or why `rballoc()`?

Reply:
Rballoc is what is used `module_load_config` to alloc the same region. Used it to reduce the number of changes.